ckanext/datastore/controller.py

```diff
@@ -43,7 +44,8 @@
 
 
 DUMP_FORMATS = 'csv', 'tsv', 'json', 'xml'
-PAGINATE_BY = 32000
+PAGINATE_BY = min(32000,
+                  int(config.get('ckan.datastore.search.rows_max', 32000)))
```

Comment:
no, leave this as a constant and instead of checking the number returned check the records_truncated value returned from datastore search in the loop below

Reply:
^ should have said "in addition to checking the number of rows returned also check the records_truncated value"

Reply:
also, trying to read a configuration value at module load time won't work when we need to reload the configuration file.

Reply:
So in the loop we check the records_truncated value and break?

I'll have to add records_truncated to datastore_search too then, I've only got it in datastore_search_sql so far.

Reply:
right I keep getting those confused

Reply:
This discussion continued in the main PR comments below ⬇️ - Dec 7

The upshot of that was that changing the interface to the dump controller was reverted. Instead, we just get dump controller's calls to datastore_search to be sensitive to the situation when rows_max < PAGINATE_BY and reduce the page size to match. So this concern is resolved.